rotkehlchen/exchanges/kraken.py

```diff
@@ -1050,3 +1145,137 @@ def history_event_from_kraken(
+        except requests.exceptions.RequestException as e:
+            raise RemoteError(f'Kraken API request failed due to {e!s}') from e
+        self._manage_call_counter(method)
+
+        return _check_and_get_response(response, method)
+
+    def query_futures_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
+        log.debug(f'querying futures balances for {self.location} with kwargs {kwargs}...')
+        raw_balances, msg = self.query_balances_method('accounts')
+        log.debug(f'got Kraken Futures raw balances = {raw_balances}')
+        if raw_balances is None:
+            return raw_balances, msg
+
+        accounts: dict = self._get_inner_dict(raw_balances, 'accounts')
+        cash: dict = self._get_inner_dict(accounts, 'cash')
+        cash_balances: dict = self._get_inner_dict(cash, 'balances')
+        flex: dict = self._get_inner_dict(accounts, 'flex')
+        flex_currencies: dict = self._get_inner_dict(flex, 'currencies')
+        log.debug(f'Kraken Futures cash balances = {cash_balances}')
+        log.debug(f'Kraken Futures flex currencies = {flex_currencies}')
+
+        self._add_single_collateral_futures_margin_to_cash_balances(accounts, cash_balances)
+
+        upper_kraken_names = defaultdict(Any, {k.upper(): v for k, v in cash_balances.items()})
+        cash_and_single_deserialized, msg = self.deserialize_kraken_balance(upper_kraken_names)
+        if msg:
+            return None, msg
+
+        flex_balances = self._parse_multi_collateral_futures_margin(flex_currencies)
+        flex_deserialized, msg = self.deserialize_kraken_balance(flex_balances)
+        if msg:
+            return None, msg
+
+        log.debug(f'deserialized Kraken Futures flex margin = {flex_deserialized}')
+        total_futures_balances = {
+            currency:
+                Balance(
+                    cash_and_single_deserialized.get(currency, Balance(ZERO, ZERO)).amount
+                    + flex_deserialized.get(currency, Balance(ZERO, ZERO)).amount,
+                    cash_and_single_deserialized.get(currency, Balance(ZERO, ZERO)).value
+                    + flex_deserialized.get(currency, Balance(ZERO, ZERO)).value,
+                    )
+            for currency in cash_and_single_deserialized.keys() | flex_deserialized.keys()
+        }
+
+        log.debug(f'done querying futures balances for {self.location}')
+        log.debug(f'total Kraken Futures balances = {total_futures_balances}')
+        return total_futures_balances, ''
+
+    @staticmethod
+    def _get_inner_dict(dictionary: dict, inner_dict_keyname: str) -> dict:
+        result: dict | None = dictionary.get(inner_dict_keyname)
+        if result is None:
+            raise RemoteError('Missing result in kraken futures response for accounts')
+
+        return result
+
+    def _parse_multi_collateral_futures_margin(
+            self, flex_currencies: dict,
+    ) -> defaultdict[Any, float]:
```

Comment:
add docstring explaining what it does. No need to say what the arguments are but explain what the function is for